core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java

```diff
@@ -382,16 +388,22 @@ protected void startServingRPCServer() {
     } catch (TTransportException e) {
       throw new RuntimeException(e);
     }
+    // create executor service for thrift.
+    // use shutdown timeout as keep-alive.
+    int stopTimeoutMs = (int) Configuration.getMs(PropertyKey.MASTER_THRIFT_SHUTDOWN_TIMEOUT);
+    mExecutorService = new ThreadPoolExecutor(mMinWorkerThreads,
```

Comment:
is this similar to how thrift would have created its own executor service internally?

Reply:
It's functionally equivalent to how thrift creates internally.